frontend-react/src/network/api/WatersApi.ts

```diff
@@ -3,24 +3,24 @@ import { Destination } from "../../resources/ActionDetailsResource";
 import { API } from "./NewApi";
 
 export class WatersResponse {
-    actualCompletionAt: string | undefined;
-    destinationCount: number | undefined;
-    destinations: Destination[] | undefined;
-    errorCount: number | undefined;
-    errors: ResponseError[] | undefined;
-    externalName: string | undefined;
-    httpStatus: number | undefined;
-    id: string | undefined;
-    overallStatus: string | undefined;
-    plannedCompletionAt: string | undefined;
-    reportItemCount: number | undefined;
-    sender: string | undefined;
-    submissionId: number | undefined;
-    timestamp: string | undefined;
-    topic: string | undefined;
-    warningCount: number | undefined;
-    warnings: ResponseError[] | undefined;
-    ok: boolean | undefined;
+    actualCompletionAt?: string;
```

Comment:
the `| undefined` amounted to making these all optional, and when creating the fixtures to use for tests, having them as optional is much easier to work with. Probably not great for these to all be optional in any case, but that can be revisited in the next ticket when we move the data fetching to the new hook

Reply:
I'd stick to matching these types with the types from the API -- you can always use `Partial<T>` to make an "all optional" version of a type. The object itself shouldn't have unnecessary undefined attributes if we can help it.

Reply:
agreed. This is just how it was defined (not sure why), and in my mind refactoring this isn't really in scope for this change. I was planning to take a closer look at it in the next PR.

Reply:
OH, I thought you'd made the object that way prior and just updated it to use `?` instead of `undefined` -- got it! Slap some kinda JSDOC on it with a `@todo`. I love that it gives us the TODO items in the IDE's quick documentation window when you hover the class!